theano/tensor/alt_sgemm.c

```diff
@@ -0,0 +1,77 @@
+/**
+C Implementation of sgemm_ based on NumPy
+Used instead of blas when Theano config flag blas.ldflags is empty.
+**/
+inline PyObject* alt_transpose(PyArrayObject* o) {
```

Comment:
This function is used only once and is very short. Remove it and inline the code manually.

idem for alt_matrix_matrix_product and alt_matrix_matrix_product2